pkg/manifests/manifests_test.go

```diff
@@ -1194,6 +1195,89 @@ ingress:
 	}
 }
 
+func TestPrometheusQueryLogFileConfig(t *testing.T) {
+	testCases := []struct {
+		name             string
+		queryLogFilePath string
+		expected         string
+		errExpected      bool
+	}{
+		{
+			name:             "basic config",
+			queryLogFilePath: "/tmp/query.log",
+			expected:         "/tmp/query.log",
+			errExpected:      false,
+		},
+		{
+			name:             "query log file on the TSDB storage volume",
+			queryLogFilePath: "/prometheus/query.log",
+			expected:         "/prometheus/query.log",
+			errExpected:      false,
+		},
+		{
+			name:             "query log file on the TSDB storage volume",
+			queryLogFilePath: "/dev/stdout",
+			expected:         "/dev/stdout",
+			errExpected:      false,
+		},
+		{
+			name:             "invalid path",
+			queryLogFilePath: "/query.log",
+			expected:         "",
+			errExpected:      true,
+		},
+		{
+			name:             "invalid file under dev",
+			queryLogFilePath: "/dev/query.log",
+			expected:         "",
+			errExpected:      true,
+		},
+	}
+	for _, tt := range testCases {
+		// We need to initialize the volumeClaimTemplate otherwise we will
+		// get a SIGSEGV when creating the EmptyDir volume
+		c, err := NewConfigFromString(`prometheusK8s:
+  volumeClaimTemplate:
+    spec:
+      resources:
+        requests:
+          storage: 15Gi
+`)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		c.ClusterMonitoringConfiguration.PrometheusK8sConfig.QueryLogFile = tt.queryLogFilePath
+		f := NewFactory("openshift-monitoring", "openshift-user-workload-monitoring", c, defaultInfrastructureReader(), &fakeProxyReader{}, NewAssets(assetsPath), &APIServerConfig{}, &configv1.Console{})
+		p, err := f.PrometheusK8s(
+			&v1.Secret{ObjectMeta: metav1.ObjectMeta{Name: "foo"}},
+			&v1.ConfigMap{ObjectMeta: metav1.ObjectMeta{Name: "foo"}},
+		)
+		if err != nil && !tt.errExpected {
+			t.Fatal(err)
+		}
+		if err == nil && tt.errExpected {
+			t.Fatal("Expected query log file ", tt.queryLogFilePath, " to give an error, but err is nil")
+		}
+		if !tt.errExpected {
+			if p.Spec.QueryLogFile != tt.expected {
+				t.Fatal("Prometheus query log is not configured correctly")
+			}
+
+			dirPath := filepath.Dir(p.Spec.QueryLogFile)
+			if dirPath != "/prometheus" && dirPath != "/dev" {
```

Comment:
It would be better to encode the check in the test case struct (e.g. `if tc.expectVolume { ... }`